go/mysql/decimal/scan.go

```diff
@@ -71,6 +72,21 @@ func parseDecimal64(s []byte) (Decimal, error) {
 	}, nil
 }
 
+// SizeAndScaleFromString
+func SizeAndScaleFromString(s string) (int32, int32) {
+	sign := 0
+	switch s[0] {
+	case '+', '-':
+		sign = 1
+	}
+	lenWithoutSign := len(s) - sign
```

Comment:
isn't this just

```suggestion
	switch s[0] {
	case '+', '-':
		s = s[1:]
	}
```

?

Reply:
Yes, you're right, that is much simpler. I don't know what I was thinking. Changed it now.